src/util/error.c

```diff
@@ -31,12 +31,11 @@ int error_set(error *e, int code, const char *msg) {
 
 int error_fmt(error *e, int code, const char *fmt, ...) {
   va_list ap;
-
   e->code = code;
+#ifdef DEBUG
   va_start(ap, fmt);
   vsnprintf(e->msg, ERROR_MSGBUF_LEN, fmt, ap);
   va_end(ap);
-#ifdef DEBUG
```

Comment:
Those messages are used by theano to report the precise cause of the error and are more generally available through gpucontext_error().

It is true that there are a couple of "regular" paths where we encounter an error so I am open to a way to improve this.

One of the way to improve this might be to spot the paths where the error can't get to the user and remove the message from those.